istio/cache/db.go

```diff
@@ -0,0 +1,76 @@
+package cache
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	log "github.com/sirupsen/logrus"
+	"github.com/stackrox/istio-cves/types"
+	"github.com/stackrox/scanner/pkg/istioutil"
+	"github.com/stackrox/scanner/pkg/vulndump"
+)
+
+var (
+	_ Cache = (*cacheImpl)(nil)
+)
+
+type cacheImpl struct {
+	cacheRWLock sync.RWMutex
+
+	cache map[string]types.Vuln
+
+	dir             string
+	timeRWLock      sync.RWMutex
+	lastUpdatedTime time.Time
+}
+
+func (c *cacheImpl) GetVulnsByVersion(vStr string) []types.Vuln {
+	c.cacheRWLock.RLock()
+	defer c.cacheRWLock.RUnlock()
+
+	var vulns []types.Vuln
+	v, err := version.NewVersion(vStr)
+	if err != nil {
+		log.Infof("Failed to get version: %s", vStr)
+		return nil
+	}
+	for _, vuln := range c.cache {
+		isAffected, _, error := istioutil.IsAffected(v, vuln)
```

Comment:
missed this before, sorry. Let's call this `err` instead, as `error` is the type